charts/latest/azuredisk-csi-driver/templates/csi-azuredisk-controller.yaml

```diff
@@ -26,7 +26,11 @@ spec:
       nodeSelector:
         kubernetes.io/os: linux
         {{- if .Values.controller.runOnMaster}}
```

Comment:
Should we add a new `runOnControlPlane` setting and have this line check if either is set and the apply the correct node-selector?

That is more work now but probably better in the long run.

Reply:
fair enough

Reply:
I think we could remove `controller.runOnMaster` chart, replace with `controller.runOnControlPlane` directly

Reply:
Won't this break back-compat for existing users? (or has a new version of the chart w/ this configuration never been released?)

Reply:
that's ok for chart config of new versions, add a new `controller.runOnControlPlane` could be safer, and after a few versions, we could remove `controller.runOnMaste`

Reply:
how about we do that as a follow-up change? in the meantime looks like we can merge this to fix broken tests

Reply:
the windows test failure should already be fixed by https://github.com/kubernetes-sigs/azuredisk-csi-driver/pull/1362, no worry, I will provide a fix using `controller.runOnControlPlane` later

Reply:
PR(https://github.com/kubernetes-sigs/azuredisk-csi-driver/pull/1366) would address the issue.